test/Microsoft.AspNetCore.Rewrite.Tests/UrlRewrite/FileParserTests.cs

```diff
@@ -0,0 +1,241 @@
+            });
+
+            var expected = new List<UrlRewriteRule>();
+            expected.Add(CreateTestRule(condList,
+                IMurl: "^article/([0-9]+)/([_0-9a-z-]+)",
+                name: "Rewrite to article.aspx",
+                UAactionType: ActionType.Rewrite,
+                UApattern: "article.aspx?id={R:1}&amp;title={R:2}"));
+            expected.Add(CreateTestRule(condList,
+                IMurl: "^article/([0-9]+)/([_0-9a-z-]+)",
+                name: "Rewrite to article.aspx",
+                UAactionType: ActionType.Redirect,
+                UApattern: "article.aspx?id={R:1}&amp;title={R:2}"));
+
+            // act
+            var res = FileParser.Parse(new StringReader(xml));
+
+            // assert
+            Assert.True(CheckUrlRewriteRuleEquality(expected, res));
+        }
+
+
+        // Creates a rule with appropriate default values of the url rewrite rule.
+        private UrlRewriteRule CreateTestRule(List<Condition> conditions,
+            LogicalGrouping condGrouping = LogicalGrouping.MatchAll,
+            bool condTracking = false,
+            string name = "",
+            bool enabled = true,
+            PatternSyntax patternSyntax = PatternSyntax.ECMAScript,
+            bool stopProcessing = false,
+            string IMurl = "",
+            bool IMignoreCase = true,
+            bool IMnegate = false,
+            ActionType UAactionType = ActionType.None,
+            string UApattern = "",
+            bool UAappendQueryString = false,
+            bool UAlogRewrittenUrl = false,
+            RedirectType UAredirectType = RedirectType.Permanent
+            )
+        {
+            return new UrlRewriteRule
+            {
+                Action = new UrlAction
+                {
+                    Url = InputParser.ParseInputString(UApattern),
+                    Type = UAactionType,
+                    AppendQueryString = UAappendQueryString,
+                    LogRewrittenUrl = UAlogRewrittenUrl,
+                    RedirectType = UAredirectType
+                },
+                Name = name,
+                Enabled = enabled,
+                StopProcessing = stopProcessing,
+                PatternSyntax = patternSyntax,
+                Match = new InitialMatch
+                {
+                    Url = new Regex(IMurl),
+                    IgnoreCase = IMignoreCase,
+                    Negate = IMnegate
+                },
+                Conditions = new Conditions
+                {
+                    ConditionList = conditions,
+                    MatchType = condGrouping,
+                    TrackingAllCaptures = condTracking
+                }
+                
+            };
+        }
+
+        private bool CheckUrlRewriteRuleEquality(List<UrlRewriteRule> expected, List<UrlRewriteRule> actual)
+        {
+            if (expected.Count != actual.Count)
+            {
+                return false;
```

Comment:
You should throw/assert here because it would give a better information to the test runner about what's wrong